File: university-timetable/src/main/java/ru/nsu/university/timetable/domain/Room.java
package ru.nsu.university.timetable.domain;

import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "rooms",
       uniqueConstraints = @UniqueConstraint(
               name = "uk_rooms_building_number",
               columnNames = {"building", "number"}))
@EntityListeners(AuditingEntityListener.class)
public class Room {
    @Id
    @GeneratedValue
    private UUID id;

    @Column(nullable = false, length = 64)
    private String building;

    @Column(nullable = false, length = 64)
    private String number;

    @Column(nullable = false)
    private int capacity;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 16)
    private Status status;

    @OneToMany(mappedBy = "room", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<RoomEquipment> equipment = new ArrayList<>();

    @CreatedDate
    @Column(nullable = false, updatable = false)
    private Instant createdAt;

    @LastModifiedDate
    @Column(nullable = false)
    private Instant updatedAt;
}

File: university-timetable-web/src/types/room.ts
import type {UUID, Instant, Status} from './common';

export interface RoomItemDto {
    name: string;
    quantity: number;
}

export interface CreateRoomRequest {
    building: string;
    number: string;
    capacity: number;
    items: RoomItemDto[];
}

export interface UpdateRoomRequest {
    building?: string;
    number?: string;
    capacity?: number;
    items?: RoomItemDto[];
}

export interface RoomResponse {
    id: UUID;
    building: string;
    number: string;
    capacity: number;
    status: Status;
    createdAt: Instant;
    updatedAt: Instant;
    items: RoomItemDto[];
}
